source/MaterialXContrib/MaterialXNode/MaterialXData.cpp

```diff
@@ -33,6 +33,13 @@ MaterialXData::MaterialXData(   mx::DocumentPtr document,
     else
     {
         _element = _document->getDescendant(elementPath);
+        if (!_element)
+        {
+            std::string message = "Element '";
+            message += elementPath;
+            message += "' not found in the document.";
```

Comment:
Is this sufficient? It's really that it wasn't found in the document, or isn't the right type, no?